### ISSUE TITLE
Batch input yields noisy images for StableDiffusionPipeline on Apple M2

### ISSUE THREAD

**Issue by n-dijkstra-enjins** [huggingface/diffusers#1519]
### Describe the bug

I am using the [StableDiffusionPipeline](https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines) from the Hugging Face [Diffusers](https://huggingface.co/docs/diffusers/index) library on an M2 Mac. When I try to generate 1 image from 1 prompt, the output looks fine, but when I try to generate multiple images in batch using the same prompt, the images are all either black squares (because of supposed NSFW content) or a noisy image (see example below). What could be the issue?

![image](https://user-images.githubusercontent.com/119493737/205253663-55a09a52-d6c5-41fe-9ec9-6a896abc3fa7.png)

NB: When I generate the images in a loop surrounding the pipe call instead of passing an iterable to the pipe call, it does work:

```
# generate images
n_imgs = 3
for i in range(n_imgs):
    img = pipe(prompt).images[0]
    # do something with img
```

### Reproduction

```
from diffusers import StableDiffusionPipeline

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
pipe = pipe.to("mps")  # for M1/M2 chips
pipe.enable_attention_slicing()

prompt = "a photo of an astronaut driving a car on mars"

# First-time "warmup" pass (because of weird M1 behaviour)
_ = pipe(prompt, num_inference_steps=1)

# generate images
n_imgs = 3 # or anything more than 1
imgs = pipe([prompt] * n_imgs).images
```

### System Info

- `diffusers` version: 0.9.0
- Platform: macOS-13.0-arm64-arm-64bit (Mac Silicon M2)
- Python version: 3.10.2
- PyTorch version (GPU?): 1.13.0 (False)
- Huggingface_hub version: 0.11.1
- Transformers version: 4.24.0
- Using GPU in script?: False
- Using distributed or parallel set-up in script?: False

**Comment by pcuenca** [comment]
Hello @n-dijkstra-enjins. I think this is because of [this PyTorch issue](https://github.com/pytorch/pytorch/issues/84039) that was mentioned in the other thread. I just pinged there again, let's hope we have some feedback soon!

**Comment by pcuenca** [comment]
Still no feedback to https://github.com/pytorch/pytorch/issues/84039, unfortunately.

**Comment by nickyreinert** [comment]
Having the exact same problem. Following the instructions here: https://huggingface.co/docs/diffusers/main/en/stable_diffusion

When generating a single image, everything works. As soon as I try to create multiple images for the same prompt, I first get the NSFW warning:
"Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed." 
and a black image.

When I disable the NSWF check:

```
pipeline = DiffusionPipeline.from_pretrained(model_id, 
                                             torch_dtype=torch.float16, 
                                             use_safetensors=True,
                                             safety_checker = None)
```

The warning is gone, but the output remains black. Someone suggested odd numbers of images. which did not work. Even the single image generation does not work anymore, I have to restart the Jupyter notebook. 

Code to reproduce

```
import tqdm as notebook_tqdm
from diffusers import DiffusionPipeline
from diffusers import DPMSolverMultistepScheduler
from diffusers.utils import make_image_grid
from PIL import Image

import torch

def get_inputs(batch_size=1):
    generator = [torch.Generator("mps").manual_seed(i) for i in range(batch_size)]
    prompts = batch_size * [prompt]
    num_inference_steps = 1

    return {"prompt": prompts, "generator": generator, "num_inference_steps": num_inference_steps}

def image_grid(imgs, rows=2, cols=2):
    w, h = imgs[0].size
    grid = Image.new("RGB", size=(cols * w, rows * h))

    for i, img in enumerate(imgs):
        grid.paste(img, box=(i % cols * w, i // cols * h))
    return grid

device = "mps" # cuda

model_id = "runwayml/stable-diffusion-v1-5"

pipeline = DiffusionPipeline.from_pretrained(model_id, 
                                             torch_dtype=torch.float16, 
                                             use_safetensors=True,
                                             safety_checker = None)

pipeline.enable_attention_slicing()

pipeline = pipeline.to(device)

pipeline.scheduler = DPMSolverMultistepScheduler.from_config(pipeline.scheduler.config)

generator = torch.Generator(device)

prompt = "portrait photo of a old warrior chief"

images = pipeline(**get_inputs(batch_size=3)).images
image_grid(images)

```

FWIWF: disabling the "_run_safety_checker_" in ```pipeline_stable_diffusion.py``` does not help, so it must come from somewhere else.
